src/app/features/automated-status-update/automated-status-update-config-service.ts

```diff
@@ -25,283 +25,370 @@ export class AutomatedStatusUpdateConfigService {
+   * Find all inheritance rules where sourceEntityType is undefined (inheritance rules)
+   * and the sourceReferenceField could potentially reference this entity type
    */
-  private processAutomatedRule(
-    targetEntityType: EntityConstructor,
-    targetFieldId: string,
-    rule: DefaultValueConfigUpdatedFromReferencingEntity,
-  ) {
-    const key = `${rule.relatedEntityType}|${rule.relatedTriggerField}`;
-    const entry = {
-      targetEntityType: targetEntityType,
-      targetFieldId: targetFieldId,
-      relatedReferenceField: rule.relatedReferenceField,
-      rule: rule,
-    };
-
-    const existingEntries = this.dependencyMap.get(key) || [];
-    this.dependencyMap.set(key, [...existingEntries, entry]);
+  private getInheritanceRulesReferencingThisEntity(
+    sourceEntityType: EntityConstructor,
+  ): Array<{
+    rule: DefaultValueConfigInheritedField;
+    targetEntityType: EntityConstructor;
+    targetFieldId: string;
+  }> {
+    const rules: Array<{
+      rule: DefaultValueConfigInheritedField;
+      targetEntityType: EntityConstructor;
+      targetFieldId: string;
+    }> = [];
+
+    for (const targetEntityType of this.entityRegistry.values()) {
+      for (const [fieldId, fieldConfig] of targetEntityType.schema.entries()) {
+        if (fieldConfig.defaultValue?.mode === "inherited-field") {
+          const rule = fieldConfig.defaultValue
+            .config as DefaultValueConfigInheritedField;
+
+          // For inheritance rules: sourceEntityType is undefined
+          if (!rule?.sourceEntityType && rule?.sourceReferenceField) {
+            // Check if the sourceReferenceField could reference our entity type
+            const referenceFieldConfig = targetEntityType.schema.get(
+              rule.sourceReferenceField,
+            );
+
+            // If the reference field is configured to reference our entity type
+            if (
+              referenceFieldConfig?.dataType === "entity" &&
+              referenceFieldConfig?.additional === sourceEntityType.ENTITY_TYPE
+            ) {
+              rules.push({
+                rule,
+                targetEntityType,
+                targetFieldId: fieldId,
+              });
+            }
+          }
+        }
+      }
+    }
+
+    return rules;
   }
 
   /**
-   * Retrieves all automated rules for a given source entity field.
-   * @param sourceEntityType - The entity type of the source
-   * @param changedField - The changed field in the source entity
+   * Load affected entities for automation rules
    */
-  public findEntitiesDependingOnField(
-    sourceEntityType: string,
-    changedField: string,
-  ) {
-    return this.dependencyMap.get(`${sourceEntityType}|${changedField}`) || [];
+  private async loadAffectedDirectEntities(
+    rules: DefaultValueConfigInheritedField[],
+    sourceEntity: Entity,
+  ): Promise<AffectedEntity[]> {
+    const affectedEntities: AffectedEntity[] = [];
+
+    for (const rule of rules) {
```

Comment:
for easier reading, maybe we can adjust it to take a single rule and return all affected entities (taking the for-loop outside of this method, so it is less nested. We in the parent caller we can maybe just flatten the results of something like `allAffected = allRules.map(r => this.loadAffectedEntities(r, entity)).flat()`